test/chemistry/test_driver_methods_gsc.py

```diff
@@ -37,18 +41,19 @@ def setUp(self):
     def _run_driver(driver, transformation=TransformationType.FULL,
                     qubit_mapping=QubitMappingType.JORDAN_WIGNER, two_qubit_reduction=False,
                     freeze_core=True):
-        qmolecule = driver.run()
 
-        core = Hamiltonian(transformation=transformation,
-                           qubit_mapping=qubit_mapping,
-                           two_qubit_reduction=two_qubit_reduction,
-                           freeze_core=freeze_core,
-                           orbital_reduction=[])
+        fermionic_transformation = \
+            FermionicTransformation(transformation=transformation,
+                                    qubit_mapping=qubit_mapping,
+                                    two_qubit_reduction=two_qubit_reduction,
+                                    freeze_core=freeze_core,
+                                    orbital_reduction=[])
+
+        solver = NumPyMinimumEigensolver()
 
-        qubit_op, aux_ops = core.run(qmolecule)
+        gsc = MinimumEigensolverGroundStateCalculation(fermionic_transformation, solver)
```

Comment:
This certainly brings in more of the stack to test the driver. Before it used the equivalent of the FermionicTransformation directly - its its interpret method took an MinEigResult (maybe as one of the types as mentioned above) that would still be doable.

Reply:
I do not agree with this comment. We need to do a ground state calculation and indeed we can use the old way of using the minimum eigensolver eigenstate calculation but the MinimumEigensolverGroundStateCalculation can handle this. The test should be ok. I can revert it to the old logic, but I am not sure it is of good use.

Reply:
This ties into a general problem which we have in the chemistry unittests: they aren't really unittests.
Most of the tests actually use the full stack end-to-end and whether we do the assembly manually here or use the MESGSC to do so for us is really not the crux of that problem.

Given that I don't think that this really matters to much. In fact, the unittests of chemistry should be reworked on a more general level to be more *unit*-oriented.

Reply:
Sometimes it is hard to really test that the unit output is valid without engaging other code. I agree it does not matter too too much here - my greater concern is the loss of flexibility of the Hamilonian replacement to directly interpret an algorithm result into a chemistry result that the Hamiltonian did in the past. That in itself would alleviate the need to pull in the MESGSC as well. But this is being discussed elsewhere...